lib/wibox/widget/slider.lua

```diff
@@ -0,0 +1,508 @@
+        or properties.handle_shape
+
+    -- Lets get the margins for the handle
+    margins = self._private.handle_margins
+        or beautiful.slider_handle_margins
+
+    x_offset, y_offset = 0, 0
+
+    if margins then
+        if type(margins) == "number" then
+            x_offset, y_offset = margins, margins
+            handle_width  = handle_width  - 2*margins
+            handle_height = handle_height - 2*margins
+        else
+            x_offset, y_offset = margins.left or 0, margins.top or 0
+            handle_width  = handle_width  -
+                (margins.left or 0) - (margins.right  or 0)
+            handle_height = handle_height -
+                (margins.top  or 0) - (margins.bottom or 0)
+        end
+    end
+
+    local value = self._private.value or self._private.min or 0
+
+    -- Get the widget size back to it's non-transfored value
+    --local matrix = gmatrix.from_cairo_matrix(cr:get_matrix())
+    local min, _, interval = get_extremums(self)
+    local rel_value = ((value-min)/interval) * (width-handle_width)
+
+    cr:translate(x_offset + rel_value, y_offset)
+
+    local handle_border_width = self._private.handle_border_width
+        or beautiful.slider_handle_border_width
+        or properties.handle_border_width or 0
+
+    handle_shape(cr, handle_width, handle_height)
+
+    if handle_border_width > 0 then
+        cr:fill_preserve()
+    else
+        cr:fill()
+    end
+
+    -- Draw the handle border
+    if handle_border_width > 0 then
+        local handle_border_color = self._private.handle_border_color
+            or beautiful.slider_handle_border_color
+            or properties.handle_border_color
+
+        if handle_border_color then
+            cr:set_source(color(handle_border_color))
+        end
+
+        cr:set_line_width(handle_border_width)
+        cr:stroke()
+    end
+end
+
+function slider:fit(_, width, height)
+    -- Use all the space, this should be used with a constraint widget
+    return width, height
+end
+
+local event = {}
+
+local rect_order = {"x", "y", "width", "height"}
+
+-- Move the handle to the correct location
+local function move_handle(self, x, y, geo)
+    -- Fix the X position when used in a rotate or mirror widget
+
+    -- Remove the translation to bring {x, y} and the rect on the same origin
+    local rem_translate = geo.matrix_to_device*gmatrix.identity
+    rem_translate.x0, rem_translate.y0 = 0, 0
+
+    local rect = {rem_translate:transform_rectangle(
+        0,
+        0,
+        geo.width,
+        geo.height
+    )}
+
+    -- Floating point precision is here both too much and not enough
+    for k, v in ipairs(rect) do
+        rect[rect_order[k]] = util.round(v)
+    end
+
+    local px = rem_translate:transform_point(x, y)
+
+    -- If the transformation moved the rectangle away from {0,0}, compensate
+    if rect.x < 0 then
+        px = rect.width + px
+    end
+
+    self:set_value(get_value(self, rect.width, px))
```

Comment:
So this function does....
- Create a matrix that translates from widget coordinates to device coordinates, but ignoring/removing the translation part of it (so just rotations and scaling is left)
- Calculates the rounded(????) extents of this widget in non-translated-device-space
- Calculates the x-coordinate of the passed-in point in non-translated device coordinates
- If this turns out to be negative, the width of the widget is added to the result

I'm not even sure which coordinate system the result is supposed to be in. Especially the "if it's negative, just add something" seems like it transforms into the black magic coordinate system.
I'd expect the following to work, but will now try it for myself and see why it apparently does not work
```lua
local function move_handle(self, x, y, geo)
    self:set_value(get_value(self, geo.width, x))
end
```